rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,184 @@ def _query_lending_balances(
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+
+                    event = HistoryBaseEntry(
+                        event_identifier=event_identifier,
+                        sequence_index=idx,
+                        timestamp=timestamp,
+                        location=self.location,
+                        location_label=self.name,
+                        asset=asset,
+                        balance=Balance(
+                            amount=interest_received,
+                            usd_value=ZERO,
+                        ),
+                        notes=f'Interest paid from {entry["lendingType"]} {entry["productName"]} savings',  # noqa: E501
+                        event_type=HistoryEventType.RECEIVE,
+                        event_subtype=HistoryEventSubType.REWARD,
+                    )
+                    events.append(event)
+
+            if len(events) != 0:
+                with self.db.user_write() as write_cursor:
+                    try:
+                        self.history_events_db.add_history_events(write_cursor, events)
+                    except InputError as e:
+                        self.msg_aggregator.add_error(
+                            f'Failed to save Binance lending interest history {query_start_ts} '
+                            f'to {query_end_ts} in database. {str(e)}',
+                        )
+
+            with self.db.user_write() as write_cursor:
+                ranges.update_used_query_range(
+                    write_cursor=write_cursor,
+                    location_string=range_query_name,
+                    queried_ranges=[(start_ts, end_ts)] + ranges_to_query,
+                )
+
+    def query_lending_interests_history_balance(
+            self,
+    ) -> dict[str, defaultdict[AssetWithOracles, Balance]]:
+        """Queries binance lending interest history and aggregate the interest gained per asset.
+        May raise:
+        - RemoteError
+        """
+        now = ts_now()
+        balances: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        interests: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        filter_query = HistoryEventFilterQuery.make(
+            from_ts=BINANCE_LAUNCH_TS,
+            to_ts=now,
+            event_types=[HistoryEventType.RECEIVE],
+            event_subtypes=[HistoryEventSubType.REWARD],
+            location=Location.BINANCE,
+            location_label=self.name,
+        )
+        with self.db.conn.read_ctx() as cursor:
+            self._query_lending_interests_history(
+                cursor=cursor,
+                start_ts=BINANCE_LAUNCH_TS,
+                end_ts=now,
+            )
+            events = self.history_events_db.get_history_events(
+                cursor=cursor,
+                filter_query=filter_query,
+                has_premium=True,
+            )
```

Comment:
since you only need the asset and the balance why not just do a `sum(cast(amout AS REAL)), sum(cast(usd_value AS REAL)) .. group by asset` and the filter you just built? I believe is the only information you are using right?